code/game/objects/structures/mirror.dm

```diff
@@ -155,7 +155,16 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 	if(!selectable_races[racechoice])
 		return TRUE
 
-	var/datum/species/newrace = selectable_races[racechoice]
+
+	var/datum/species/newrace = new selectable_races[racechoice]
+
+	var/attributes_desc = newrace.get_physical_attributes()
```

Comment:
```suggestion
	var/attributes_desc = newrace.get_physical_attributes()
	qdel(newrace)
```

otherwise we're just initializing datums and leaving them to rot and do nothing. Is there a global list of initialized species datums that we can access for stuff like this, so we don't have to waste time? Given how many magic mirrors are used I don't think it necessitates making something like that, but it would be nice to use something like that rather than have to init/qdel a datum every time.